src/SchemaGenerator.ts

```diff
@@ -135,6 +135,9 @@ export class SchemaGenerator {
             const name = child.getName();
             if (!(name in definitions)) {
                 definitions[name] = this.typeFormatter.getDefinition(child.getType());
+                if (this.config?.definitionTitles) {
```

Comment:
I'm not sure if this is the best place. Something inside `getDefinition` should be better. This doesn't work for deep objects/definitions, either.